Hazelcast.Test/Hazelcast.NearCache.Test/NearcacheTestSupport.cs

```diff
@@ -0,0 +1,142 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using Hazelcast.Client.Proxy;
+using Hazelcast.Client.Test;
+using Hazelcast.Core;
+using Hazelcast.IO.Serialization;
+using Hazelcast.Remote;
+
+namespace Hazelcast.NearCache.Test
+{
+    public class NearcacheTestSupport : MultiMemberBaseTest
+    {
+        internal BaseNearCache GetNearCache<TK, TV>(IMap<TK, TV> map)
+        {
+            var mapProxy = (map as ClientMapNearCacheProxy<TK, TV>);
+            return mapProxy != null ? mapProxy.NearCache : null;
+        }
+
+        protected Response RemoveKeyAtServer(string map, string key)
+        {
+            var script = string.Format(@"map=instance_0.getMap('{0}')
+map.remove('{1}')
+", map, key);
+            return RemoteController.executeOnController(HzCluster.Id, script, Lang.PYTHON);
+        }
+
+        protected Response PutOnMember(int key, int value, string map)
+        {
+            var script = string.Format(@"map=instance_0.getMap('{0}')
+map.put({1}, {2})
+", map, key, value);
+            return RemoteController.executeOnController(HzCluster.Id, script, Lang.PYTHON);
+        }
+
+        protected Response PopulateMapFromServer(string mapName, int mapSize)
+        {
+            var script = string.Format(@"map=instance_0.getMap('{0}')
+for i in xrange(0, {1}):
+    map.put(i, i)
+", mapName, mapSize);
+            return RemoteController.executeOnController(HzCluster.Id, script, Lang.PYTHON);
+        }
+
+        protected Response PopulateMapWithRandomValueFromServer(string mapName, int mapSize)
+        {
+            var script = string.Format(@"
+from com.hazelcast.util import RandomPicker
+map=instance_0.getMap('{0}')
+for i in xrange(0, {1}):
+    map.put(i, RandomPicker.getInt( {1}))
+", mapName, mapSize);
+            return RemoteController.executeOnController(HzCluster.Id, script, Lang.PYTHON);
+        }
+
+        protected List<object> GetAllValueFromMember(int mapSize, string mapName)
+        {
+            var script = string.Format(@"
+from java.util import ArrayList
+map=instance_0.getMap('{0}')
+list= ArrayList()
+for i in xrange(0, {1}):
+    list.add(map.get(i))
+result=instance_0.getSerializationService().toBytes(list)
+", mapName, mapSize);
+            var response = RemoteController.executeOnController(HzCluster.Id, script, Lang.PYTHON);
+            if (!response.Success)
+            {
+                throw new InvalidOperationException(response.ToString());
+            }
+            var data = new HeapData(response.Result);
+            return ClientInternal.GetSerializationService().ToObject<List<object>>(data);
+        }
+
+        protected Response DistortRandomPartitionUuid()
+        {
+            const string script = @"
```

Comment:
We have this entry processor in test package: https://github.com/hazelcast/hazelcast/blob/master/hazelcast/src/test/java/com/hazelcast/client/test/DistortInvalidationMetadataEntryProcessor.java
It could simplify the tests here but I am ok with either of these.